File: examples/ts-calc/calc.ts
// A tiny calculator in TypeScript, for the TS verifiable-mastery demo.

export class Calculator {
  add(a: number, b: number): number {
    return a + b;
  }

  sub(a: number, b: number): number {
    return a - b;
  }

  addMany(nums: number[]): number {
    let total = 0;
    for (const n of nums) {
      total = this.add(total, n);
    }
    return total;
  }
}

export function average(nums: number[]): number {
  if (nums.length === 0) return 0;
  return new Calculator().addMany(nums) / nums.length;
}

File: examples/mixed-app/calc.py
"""A tiny calculator used by the mixed code+docs demo."""


class Calculator:
    def add(self, a, b):
        return a + b

    def add_many(self, nums):
        total = 0
        for n in nums:
            total = self.add(total, n)
        return total


def average(nums):
    if not nums:
        return 0
    return Calculator().add_many(nums) / len(nums)
